### PR TITLE
added params support to id seek

### PR DESCRIPTION
**Pull request by DvirDukhan**

### CODE REVIEW

File: src/execution_plan/optimizations/seek_by_id.c [hunk 1]
```diff
@@ -24,6 +24,10 @@ static bool _idFilter(FT_FilterNode *f, AST_Operator *rel, EntityID *id, bool *r
 	AR_ExpNode *rhs = f->pred.rhs;
 	*rel = f->pred.op;
 
+	// In case of parameter, evalute it so it will in-place become constant.
+	if(AR_EXP_IsParameter(lhs)) AR_EXP_Evaluate(lhs, NULL);
+	if(AR_EXP_IsParameter(rhs)) AR_EXP_Evaluate(rhs, NULL);
```
Comment (swilly22) [inline src/execution_plan/optimizations/seek_by_id.c:29]: Can't we change this to:

```
AR_EXP_Evaluate(lhs, NULL);
AR_EXP_Evaluate(rhs, NULL);
```

Ignoring errors ?

File: src/execution_plan/optimizations/utilize_indices.c [hunk 2]
```diff
@@ -218,11 +218,11 @@ static bool _validateInExpression(AR_ExpNode *exp) {
 	assert(exp->op.child_count == 2);
 
 	AR_ExpNode *list = exp->op.children[1];
-	// In case the array is a parameter such as "WHERE x IN $arr", evaluate the parameter for in-place replacement.
+	SIValue listValue = SI_NullVal();
+	bool constant_list = AR_EXP_ReduceToScalar(list, true, &listValue);
```
Comment (swilly22) [inline src/execution_plan/optimizations/utilize_indices.c:222]: Please rename `constant_list` to `reduced`

File: src/execution_plan/optimizations/utilize_indices.c [hunk 3]
```diff
@@ -218,11 +218,11 @@ static bool _validateInExpression(AR_ExpNode *exp) {
...
 	if(AR_EXP_IsParameter(list)) AR_EXP_Evaluate(list, NULL);
-	if(list->operand.type != AR_EXP_CONSTANT || list->operand.constant.type != T_ARRAY) return false;
+	if(!constant_list || SI_TYPE(listValue) != T_ARRAY) return false;
```
Comment (swilly22) [inline src/execution_plan/optimizations/utilize_indices.c:224]: I think checking if `SI_TYPE(listValue) != T_ARRAY` should be enough.

File: src/arithmetic/arithmetic_expression.c [hunk 4]
```diff
@@ -172,10 +172,17 @@ AR_ExpNode *AR_EXP_NewParameterOperandNode(const char *param_name) {
  * e.g. MINUS(X) where X is a constant number will be reduced to
  * a single node with the value -X
  * PLUS(MINUS(A), B) will be reduced to a single constant: B-A. */
-bool AR_EXP_ReduceToScalar(AR_ExpNode *root) {
+bool AR_EXP_ReduceToScalar(AR_ExpNode *root, bool runtime, SIValue *val) {
```
Comment (swilly22) [inline src/arithmetic/arithmetic_expression.c:175]: I think `val` should be set to NULL on entry and set to scalar only of the entire expression was reduced,
I know this is a recursive function, but you might want to introduce a  `static bool _ AR_EXP_ReduceToScalar` function to help out.

File: src/execution_plan/optimizations/utilize_indices.c (hunk 3 above)
Comment (DvirDukhan) [reply src/execution_plan/optimizations/utilize_indices.c:224]: I prefer this for my peace of mind.

File: src/arithmetic/arithmetic_expression.c (hunk 4 above)
Comment (DvirDukhan) [reply src/arithmetic/arithmetic_expression.c:175]: setting to NullVal

**Pull request merged**